packages/candid/src/idl.ts

```diff
@@ -774,7 +780,37 @@ export class VecClass<T> extends ConstructType<T[]> {
     }
     const len = Number(lebDecode(b));
     if (this._blobOptimization) {
-      return [...new Uint8Array(b.read(len))] as unknown as T[];
+      return new Uint8Array(b.read(len)) as unknown as T[];
+    }
+
+    if (this._type instanceof FixedNatClass) {
+      if (this._type.bits == 8) {
+        return new Uint8Array(b.read(len)) as unknown as T[];
```

Comment:
Is this a breaking change? If so, we need to document it.

Reply:
I added a doc string to VecClass. 

Not sure how to add to CHANGELOG.md since there is none in this repo.